NuRadioReco/modules/analogToDigitalConverter.py

```diff
@@ -0,0 +1,318 @@
+            trace = diode.tunnel_diode(channel)
+
+        if trigger_adc:
+            adc_time_delay_label = "trigger_adc_time_delay"
+            adc_n_bits_label = "trigger_adc_nbits"
+            adc_ref_voltage_label = "trigger_adc_reference_voltage"
+            adc_sampling_frequency_label = "trigger_adc_sampling_frequency"
+        else:
+            adc_time_delay_label = "adc_time_delay"
+            adc_n_bits_label = "adc_nbits"
+            adc_ref_voltage_label = "adc_reference_voltage"
+            adc_sampling_frequency_label = "adc_sampling_frequency"
+
+        adc_time_delay = 0
+
+        if adc_time_delay_label in det_channel:
+            if det_channel[adc_time_delay_label] is not None:
+                adc_time_delay = channel[adc_time_delay_label] * units.ns
+
+        if random_clock_offset:
+            clock_offset = np.random.uniform(-1, 1)
+            adc_time_delay += clock_offset / channel.get_sampling_rate()
+
+        adc_n_bits = det_channel[adc_n_bits_label]
+        adc_ref_voltage = det_channel[adc_ref_voltage_label] * units.V
+        adc_sampling_frequency = det_channel[adc_sampling_frequency_label] * units.GHz
+
+        if (adc_sampling_frequency > channel.get_sampling_rate()):
+            error_msg  = 'The ADC sampling rate is greater than '
+            error_msg += 'the channel {} sampling rate. '.format(channel.get_id())
+            error_msg += 'Please change the ADC sampling rate.'
+            raise ValueError(error_msg)
+
+        delayed_times = times + adc_time_delay
+        interpolate_trace = interp1d(times, trace, kind='quadratic',
+                                     fill_value='extrapolate')
+
+        delayed_trace = interpolate_trace(delayed_times)
+
+        new_n_samples = int( (adc_sampling_frequency / channel.get_sampling_rate()) * len(delayed_trace) )
+        resampled_trace = resample(delayed_trace, new_n_samples)
+
+        digital_trace = self._adc_types[adc_type](delayed_trace, adc_n_bits,
+                                                  adc_ref_voltage, output)
+
+        if return_sampling_frequency:
+            return digital_trace, adc_sampling_frequency
+        else:
+            return digital_trace
+
+    @register_run()
+    def run(self, evt, station, det,
+            random_clock_offset=True,
+            adc_type='perfect_floor_comparator',
+            output='voltage'):
+        """
+        Runs the analogToDigitalConverter and transforms the traces from all
+        the channels of an input station to digital voltage values.
+
+        Parameters
+        ----------
+        evt: framework.event.Event object
+        station: framework.station.Station object
+        det: detector.detector.Detector object
+        random_clock_offset: bool
+            If True, a random clock offset between -1 and 1 clock cycles is added
+        adc_type: string
+            The type of ADC used. The following are available:
+            - perfect_floor_comparator
```

Comment:
Doesn't it also allow perfect_ceil_comparator?

Reply:
Changed